Update docs (#100)
dls-controls/tickit: docs/user/explanations/components.rst

Simulation Components
=====================

There are two types of components that can be used in a tickit simulation,
device simulations and system simulations.

Device Simulation
-----------------

Device simulations are the typical use case of a component. They encapsulate a
device and any adapters for that device.

.. figure:: ../../images/tickit-device-simulation-cpt.svg
    :align: center

(See `DeviceSimulation`.)

System Simulation
-----------------

System simulation components are themselves entire tickit simulations. They
contain their own device simulation components and a scheduler for orchestrating
them. However, the scheduler in a system component acts as a slave scheduler
which is driven by the master scheduler in the top level of the simulation
it belongs to.

.. figure:: ../../images/tickit-system-simulation-cpt.svg
    :align: center

System simulations can also contain their own system simulation components
allowing for the construction of reasonably complex systems.

> To configure a system simulation the config.yaml the system simulation components
> and their respective components are nested in the config.yaml in order to
> populate the wiring for the master scheduler. For example:

Reviewer: ```suggestion
System simulations can be nested inside other components in the config so that the master scheduler's wiring is correct, for example: 
```
